lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -385,284 +388,310 @@ func (c *Client) startInputStreaming(stopCh chan struct{}) error {
-					completion_id: C.uint32_t(m.CompletionID),
-					err_code:      m.ErrCode,
-				}); errCode != C.ErrCodeSuccess {
-					return trace.Errorf("SharedDirectoryMoveResponse failed: %v", errCode)
-				}
+		}
+	case tdp.SharedDirectoryWriteResponse:
+		if c.cfg.AllowDirectorySharing {
+			if errCode := C.client_handle_tdp_sd_write_response(C.ulong(c.handle), C.CGOSharedDirectoryWriteResponse{
+				completion_id: C.uint32_t(m.CompletionID),
+				err_code:      m.ErrCode,
+				bytes_written: C.uint32_t(m.BytesWritten),
+			}); errCode != C.ErrCodeSuccess {
+				return trace.Errorf("SharedDirectoryWriteResponse failed: %v", errCode)
 			}
-		case tdp.SharedDirectoryTruncateResponse:
-			if c.cfg.AllowDirectorySharing {
-				if errCode := C.client_handle_tdp_sd_truncate_response(C.ulong(c.handle), C.CGOSharedDirectoryTruncateResponse{
-					completion_id: C.uint32_t(m.CompletionID),
-					err_code:      m.ErrCode,
-				}); errCode != C.ErrCodeSuccess {
-					return trace.Errorf("SharedDirectoryTruncateResponse failed: %v", errCode)
-				}
+		}
+	case tdp.SharedDirectoryMoveResponse:
+		if c.cfg.AllowDirectorySharing {
+			if errCode := C.client_handle_tdp_sd_move_response(C.ulong(c.handle), C.CGOSharedDirectoryMoveResponse{
+				completion_id: C.uint32_t(m.CompletionID),
+				err_code:      m.ErrCode,
+			}); errCode != C.ErrCodeSuccess {
+				return trace.Errorf("SharedDirectoryMoveResponse failed: %v", errCode)
 			}
-		case tdp.RDPResponsePDU:
-			pduLen := uint32(len(m))
-			if pduLen == 0 {
-				c.cfg.Logger.ErrorContext(context.Background(), "response PDU empty")
+		}
+	case tdp.SharedDirectoryTruncateResponse:
+		if c.cfg.AllowDirectorySharing {
+			if errCode := C.client_handle_tdp_sd_truncate_response(C.ulong(c.handle), C.CGOSharedDirectoryTruncateResponse{
+				completion_id: C.uint32_t(m.CompletionID),
+				err_code:      m.ErrCode,
+			}); errCode != C.ErrCodeSuccess {
+				return trace.Errorf("SharedDirectoryTruncateResponse failed: %v", errCode)
 			}
-			rdpResponsePDU := (*C.uint8_t)(unsafe.SliceData(m))
+		}
+	case tdp.RDPResponsePDU:
+		pduLen := uint32(len(m))
+		if pduLen == 0 {
+			c.cfg.Logger.ErrorContext(context.Background(), "response PDU empty")
+		}
+		rdpResponsePDU := (*C.uint8_t)(unsafe.SliceData(m))
 
-			if errCode := C.client_handle_tdp_rdp_response_pdu(
-				C.ulong(c.handle), rdpResponsePDU, C.uint32_t(pduLen),
-			); errCode != C.ErrCodeSuccess {
-				return trace.Errorf("RDPResponsePDU failed: %v", errCode)
-			}
-		default:
-			c.cfg.Logger.WarnContext(context.Background(),
-				fmt.Sprintf("Skipping unimplemented TDP message type %T", msg))
+		if errCode := C.client_handle_tdp_rdp_response_pdu(
+			C.ulong(c.handle), rdpResponsePDU, C.uint32_t(pduLen),
+		); errCode != C.ErrCodeSuccess {
+			return trace.Errorf("RDPResponsePDU failed: %v", errCode)
 		}
+	default:
+		c.cfg.Logger.WarnContext(context.Background(),
+			fmt.Sprintf("Skipping unimplemented TDP message type %T", msg))
```

Comment:
We no longer need to use Sprintf here. Forrest made a `TypeAttr` utility:
```suggestion
		c.cfg.Logger.WarnContext(context.Background(), "Skipping unimplemented TDP message", "type", logutils.TypeAttr(msg)))
```